Design tweaks 3.1

iainkirkpatrick opened the pull request:
Latest design tweaks from @iainkirkpatrick and @bsides44 

Let me know if there's anything unclear / I can do and I'll fix on Wednesday NZT 😄 

Deployed at https://hrmi-design-tweaks-3-1.firebaseapp.com

tmfrnz commented:
A few things that I am noticing 

### Home

- [ ] Intro images are stretched horizontally on wide screens - there must be a limit to how far these images can stretch. I had discussed this with @adi-cam that this limit could be at 1440px

![image](https://user-images.githubusercontent.com/5811956/85394189-7ac48f00-b54e-11ea-8a69-c9a8c251b745.png)

### Header

- [ ] Dropdown arrow should be the same size, it appears that only the down arrow has been reduced in the size, but not the up arrow, this also leads to a horizontal jump when opening/closing

![image](https://user-images.githubusercontent.com/5811956/85394572-0f2ef180-b54f-11ea-9368-fdd35146abfa.png)

- [ ] Language dropdown width issue

![image](https://user-images.githubusercontent.com/5811956/85394745-52896000-b54f-11ea-9f78-e44cc7db1216.png)

tmfrnz commented:
... continuing

### Header

- [ ] Search field without border or background area - could be intentional but a bit funny as inconsistent with other search fields


### Home

- [ ] Our Data section: download link should also go to WordPress site (as the header link) - instead goes to current download data page

### Header 

- [ ] Download Data button not aligned with other menu items

tmfrnz commented:
# Mobile

### Home

- [ ] slider cards very high (minor)

![image](https://user-images.githubusercontent.com/5811956/85399427-13f7a380-b557-11ea-8848-53aaed6bfb27.png)

### Charts

- [ ] x-axis chart label too wide for mobile

![image](https://user-images.githubusercontent.com/5811956/85400372-99c81e80-b558-11ea-9c48-26c5b6224660.png)

iainkirkpatrick commented:
Addressing your points @tmfrnz:

>  Intro images are stretched horizontally on wide screens - there must be a limit to how far these images can stretch. I had discussed this with @adi-cam that this limit could be at 1440px

I'm reverting this - I can see what was done was mistaken in reading Adina's notes, but I'm not confident of a working solution in the short-term

>  Download Data button not aligned with other menu items

This looks to me to be the same on the current live site - I'm wavering between seeing it not aligned and then not sure if it's just my eyes playing tricks on me. From a look in the inspector there doesn't appear to be any reason they aren't aligned... something IMO to look at later.

> slider cards very high (minor)

I believe this was the solution presented for the text causing different card heights (esp with different languages) - perhaps could use some tweaking later, but for now seems fine?

>  x-axis chart label too wide for mobile (single metric - works on single country)

I haven't seen any mobile designs for this... IMO there's a few issues with general legibility + clarity of the data on mobile, especially narrow mobile screens. We could remove the label on mobile, but that's probably not really what we want?

All the other points that I haven't quoted I've also now pushed fixes for - should be a single commit for each of them (in order). So for me, the main one is the last quoted point above re x-axis chart labels on mobile - is this a dealbreaker to getting the changes in? Happy to discuss / work on this tonight with you @tmfrnz 🙂

iainkirkpatrick commented:
Also just found that the message "FAQs" now appearing in the aside has no translations associated with it - I've removed these for now so it defaults to "FAQs" across all locales.

annemariebrook commented:
Thanks very much @iainkirkpatrick and @tmfrnz for identifying and ironing out these issues.

annemariebrook commented:
Just a couple of things I noticed:
1.     Download Data button not aligned with other menu items
This looks non-aligned to me. But agree that it's not a major and can be addressed later. 
2.   On the Safety and Empowerment tabs on the country pages, the "Source" note is very close to the bottom of the charts. If the gap could be increased slightly (to be more like that for Quality of Life), that would be great. But not a major.

tmfrnz commented on app/translations/es.json:
```diff
@@ -171,7 +171,6 @@
   "hrmi.containers.Settings.hi.title": "NO UTILIZADO: País de ingresos altos",
   "hrmi.containers.Settings.hi.text": "NO UTILIZADO: Para estos países es mejor utilizar el estándar de evaluación de ingresos altos",
   "hrmi.containers.Settings.hi.textIndicator": "NO UTILIZADO: para estos países es mejor utilizar los indicadores empleados para el estándar de evaluación de ingresos altos",
-  "hrmi.containers.FAQs.title": "TODO",
```
Rather than removing these I strongly suggest keeping them with the default value.

This will allow text editors  to add the translation(s) when they can and also ensure the same number of lines in each language file

tmfrnz replied:
fixing this now

tmfrnz commented:
for this

> x-axis chart label too wide for mobile (single metric - works on single country)

I would suggest to simply remove the worse/better annotation that is overlapping

tmfrnz commented:
> Addressing your points @tmfrnz:
> 
> > Intro images are stretched horizontally on wide screens - there must be a limit to how far these images can stretch. I had discussed this with @adi-cam that this limit could be at 1440px
> 
> I'm reverting this - I can see what was done was mistaken in reading Adina's notes, but I'm not confident of a working solution in the short-term

Yes this has always been a challenging layout proposition so will have to introduce a limit somewhere where images stop aligning with the screen edge

> 
> > Download Data button not aligned with other menu items
> 
> This looks to me to be the same on the current live site - I'm wavering between seeing it not aligned and then not sure if it's just my eyes playing tricks on me. From a look in the inspector there doesn't appear to be any reason they aren't aligned... something IMO to look at later.

fixed

> 
> > slider cards very high (minor)
> 
> I believe this was the solution presented for the text causing different card heights (esp with different languages) - perhaps could use some tweaking later, but for now seems fine?

Sure

> 
> > x-axis chart label too wide for mobile (single metric - works on single country)
> 
> I haven't seen any mobile designs for this... IMO there's a few issues with general legibility + clarity of the data on mobile, especially narrow mobile screens. We could remove the label on mobile, but that's probably not really what we want?

So actually the issue is the overlap with underlying better/worse labels, so have removed them for mobile. However one could consider removing them altogether as the new axis label kind of conveys the same message

> 
> All the other points that I haven't quoted I've also now pushed fixes for - should be a single commit for each of them (in order). So for me, the main one is the last quoted point above re x-axis chart labels on mobile - is this a dealbreaker to getting the changes in? Happy to discuss / work on this tonight with you @tmfrnz 🙂

I think this is good to go now - have a few other little things to address today so can address any (small) potential issues that have not yet been identified

tmfrnz commented:
> Just a couple of things I noticed:
> 
> 1. ```
>    Download Data button not aligned with other menu items
>    ```
> 
> This looks non-aligned to me. But agree that it's not a major and can be addressed later.

Fixed =)

> 2. On the Safety and Empowerment tabs on the country pages, the "Source" note is very close to the bottom of the charts. If the gap could be increased slightly (to be more like that for Quality of Life), that would be great. But not a major.

Might be able to have a look later...

tmfrnz commented:
@iainkirkpatrick 

Haven't noticed this before

![image](https://user-images.githubusercontent.com/5811956/85549679-84b6c280-b620-11ea-87db-77e86c076400.png)

Quite ugly I think, so for now I am changing this to 

![image](https://user-images.githubusercontent.com/5811956/85552509-5981a280-b623-11ea-8f22-9b055c3d02db.png)

- align key right
- show 'symbol' on right also (after all this is where it is shown also
- do not wrap source
- use common label (replacing hard-coded text labels

https://github.com/HumanRightsMeasurementInitiative/hrmi-dataportal/commit/f8e5ec7750e0e5a3cf4c5b276ff74a443e5337b9

tmfrnz commented:
@iainkirkpatrick, some more issues coming up so I will keep documenting here

- [ ] do not use borderSize, borderColor as attributes (instead use styled-components or style attribute)

> react_devtools_backend.js:6 Warning: React does not recognize the `borderSize` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `bordersize` instead. If you accidentally passed it from a parent component, remove it from the DOM element.

> react_devtools_backend.js:6 Warning: React does not recognize the `borderColor` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `bordercolor` instead. If you accidentally passed it from a parent component, remove it from the DOM element.

tmfrnz commented:
> @iainkirkpatrick, some more issues coming up so I will keep documenting here
> 
> * [x]  do not use borderSize, borderColor as attributes (instead use styled-components or style attribute)
> 
> > react_devtools_backend.js:6 Warning: React does not recognize the `borderSize` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `bordersize` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
> 
> > react_devtools_backend.js:6 Warning: React does not recognize the `borderColor` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `bordercolor` instead. If you accidentally passed it from a parent component, remove it from the DOM element.

resolved: https://github.com/HumanRightsMeasurementInitiative/hrmi-dataportal/commit/1676486fccf4fce265235561f398dfdaa2dcf544

tmfrnz commented:
Adding some IE 11 issues here (that I will not be able to address today unfortunately)

- logo image stretches vertically

![image](https://user-images.githubusercontent.com/5811956/85598933-551daf80-b64c-11ea-9960-bb0634db7be3.png)

also see https://github.com/HumanRightsMeasurementInitiative/hrmi-dataportal/issues/146

annemariebrook commented:
> Quite ugly I think, so for now I am changing this to
> ![image](https://user-images.githubusercontent.com/5811956/85552509-5981a280-b623-11ea-8f22-9b055c3d02db.png)
>     * align key right
>     * show 'symbol' on right also (after all this is where it is shown also
>     * do not wrap source
>     * use common label (replacing hard-coded text labels

Thanks @tmfrnz. Looks good.
